q2_metadata/normalization/_norm_check_rules.py

```diff
@@ -0,0 +1,311 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from q2_metadata.normalization._norm_errors import (
+    RuleError, ExpectedError, OntologyError, RemapError,
+    ValidationError, NormalizationError,
+    BlankError, MissingError, FormatError
+)
+
+
+def check_rule(variable: str, rules: dict, rule: str, rule_value):
+    """For each rule in the current variable's rules set,
+    perform checks that the rule values to be used for the rules
+    to apply are properly formatted.
+        - If properly formatted: full the rules dict
+        for the variable's Rules() class instance.
+        - If not properly formatted: raise a specific error and
+        collect the encountered error in the ErrorsCollection()
+        instance for this variable's Rules() class instance.
+
+    Parameters
+    ----------
+    variable : str
+        Name of the metadata variable for which there is rules.
+    rules : dict
+        Rules in the hard-coded data structure that has
+        common default values for all variables and
+        that is checked and updated here.
+    rule : str
+        Current rule name. Could be one of these:
+        - expected: accepted, controlled vocabulary from AGP.
+        - ontology: accepted, controlled vocabulary from a third-party resource.
+        - remap: the replacements to perform ( "A" -> "B" )
+        - validation: the
+        - normalization:
+        - blank:
+        - missing:
+        - format:
```

Comment:
The documentation here looks a little bit incomplete would you mind updating? Also make sure docstrings don't contain information that's repeated elsewhere. For this argument it might be better to just list the allowed values. No need to re-document what each are:

```suggestion
    rule : str
        Current rule name. Can be one of "expected", "ontology", "remap", "validation",
        "normalization", "blank", "missing" or "format".
```

If you can think of other places where the documentation could be less redundant/verbose that would be great to update..

Reply:
Okay - my understanding was that more is better in docstrings since I thought that every function's docstring could eventually be read independently from the other functions. Will do.

Reply:
More is better if and only if it is accurate and readable

Reply:
sure - what is the best practice for docstring'ing in code where objects/variables are passed through classes, methods etc? i.e. shall the object be described only the first time it is invoked? 
- what if it is an empty object the first time it is invoked ?
-> should if be described as it is at this point (i.e. empty)?
- or should the object be re-describing later in the docstring of another function/method where it is _actually_ "shaped" / "filled-up"?
Thanks for you expert advice.

Reply:
as per @ElDeveloper, there is now a general module-level doctring (see at the beginning of `_norm_rules.py`) and then the attributes / methods are described more specifically without redundancy